acmwebsite/lib/card.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+from kajiki import FileLoader, XMLTemplate
+from itertools import chain
+from acmwebsite.lib import helpers
+
+
+default_loader = FileLoader('acmwebsite/templates/cards')
+default_loader.extension_map['xhtml'] = XMLTemplate
+
+
+class CardTypes:
+    """Maintains a list of card generators. Any page can use cards by instantiating
+    a :class:`CardTypes` variable.
+    """
+
+    def __init__(self):
+        self.list = []
+
+    def register(self, generator, title_template=None, body_template=None, loader=default_loader):
+        """Register a new card generator
+
+        :param generator: a generator function that yields individual cards.
+        :param title_template: (optional) the file where the Kajiki template
+            for the cards' title is. Defaults to ``None``.
+        :param body_template: (optional) the file where the Kajiki template
+            for the cards' body is. Defaults to ``None``.
+        :param loader: (optional) a Kahjiki :class:`FileLoader` object that is
+            used to load the aforementioned template files. Defaults to a
+            :class:`FileLoader` object in the 'acmwebsite/templates/cards'
+            directory
+        :return: ``None``
+        :rtype: None
+        """
+
+        if title_template is not None:
+            title_template = loader.load(f'{title_template}.xhtml')
+
+        if body_template is not None:
+            body_template = loader.load(f'{body_template}.xhtml')
+
+        def _gen(*args, **kwargs):
```

Comment:
What does this function do? Just because it is private does not give it a good reason to have an insensible name.

Can it be placed inside of another method where it is used?